packages/runtime/src/useSSRComputation_Client.ts

```diff
@@ -0,0 +1,150 @@
+  module?: SSRComputationModule<TResult>,
+  currentResult: TResult | null,
+  importFn: ClientComputationFunction<TResult>,
+  currentExecutor?: ComputationExecutor<TResult>,
+};
+
+class ComputationExecutor<TResult> {
+  isDisposed = false;
+  subscription: Subscription | undefined;
+
+  constructor(
+    private readonly dependencies: Dependency[],
+    private readonly updateResult: (TResult) => void,
+    private readonly state: ClientState<TResult>,
+  ) {}
+
+  handleSubscriptionIfModuleLoaded = ({ recomputeTheResult }: { recomputeTheResult: boolean }) => {
+    if (this.isDisposed || !this.state.module) return;
+
+    const updateResult = (newResult: TResult) => {
+      if (this.isDisposed || this.state.currentResult === newResult) return;
+      this.updateResult(newResult);
+    }
+
+    if (recomputeTheResult) {
+      const fn = this.state.module?.compute;
+      if (fn) {
+        updateResult(fn(...this.dependencies));
+      }
+    }
+    const getCurrentResult = () => this.state.currentResult!;
+    this.subscription = this.state.module?.subscribe?.(getCurrentResult, updateResult, ...this.dependencies);
+  }
+
+  getResultIfModuleLoaded = () => {
+    const fn = this.state.module?.compute;
+    return fn ? fn(...this.dependencies) : NoResult;
+  }
+
+  loadAndRun = () => {
+    if (this.isDisposed) return;
+    this.state.importFn().then(module => {
+      if (this.isDisposed) return;
+      if (!module.compute) throw new Error('The SSR Computation module must have a compute function');
+      this.state.module = module;
+      this.handleSubscriptionIfModuleLoaded({ recomputeTheResult: true });
+    });
+  }
+
+  dispose = () => {
+    if (this.isDisposed) return;
+    this.isDisposed = true;
+    this.subscription?.unsubscribe();
+    this.subscription = undefined;
+  }
+}
+
+const useSSRComputation_Client = <TResult>(
+  importFn: ClientComputationFunction<TResult>,
+  dependencies: Dependency[],
+  options: Options,
+  relativePathToCwd: string
+): TResult | null => {
+  const [, forceUpdate] = useState(0);
+  const clientState = useRef<ClientState<TResult>>({
+    importFn,
+    currentResult: null,
+  });
+
+  const cache = getSSRCache();
+  const parsedDependencies = parseDependencies(dependencies);
+  const skip = !!options.skip;
+  // relativePathToCwd is used to make sure that the cache key is unique for each module
+  // and it's not affected by the file that calls it
+  const cacheKey = calculateCacheKey(relativePathToCwd, parsedDependencies);
+
+  const updateResult = useCallback((newResult: TResult, rerender: boolean) => {
+    cache[cacheKey] = {
+      result: newResult,
+      isSubscription: cache[cacheKey]?.isSubscription || !!clientState.current?.module?.subscribe,
+    }
+    clientState.current.currentResult = newResult;
+    if (rerender) {
+      forceUpdate(prevState => prevState + 1);
+    }
+  }, [cacheKey]);
+
+  const executor = useMemo(() => {
+    clientState.current.currentExecutor?.dispose();
```

Comment:
We create and dispose the `executor` inside `useMemo` instead of `useEffect` because we need to calculate the `currentResult` at the same render cycle if the module is already loaded.